### PR TITLE
MAISTRA-535 Add istio-cni RPM

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: istio-cni/istio-cni.spec [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+Name:           istio-cni
+Version:        0.12.0
+Release:        1%{?dist}
+Summary:        Istio CNI Plugin
+License:        ASL 2.0
+URL:            https://%{provider_prefix}
+
+Source0:        https://%{provider_prefix}/archive/%{git_commit}/%{repo}-%{git_commit}.tar.gz
+
+# e.g. el6 has ppc64 arch without gcc-go, so EA tag is required
```
Comment (brian-avery) [inline istio-cni/istio-cni.spec:24]: el6?

File: istio-cni/istio-cni.spec [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+mkdir -p istio-cni/src/%{provider_prefix}
+tar zxf %{SOURCE0} -C istio-cni/src/%{provider_prefix} --strip=1
+
+%build
+cd istio-cni
+export GOPATH=$(pwd):%{gopath}
+
+pushd src/%{provider_prefix}
+make build ISTIO_OUT=bin ISTIO_CNI_RELPATH=github.com/maistra/istio-cni
+# make VERSION=%{version} GITREVISION=%{git_shortcommit} GITSTATUS=Clean GITTAG=%{version}-%{release}
```
Comment (brian-avery) [inline istio-cni/istio-cni.spec]: Do we need this commented code?

**Review by brian-avery — APPROVED** [review]

File: istio-cni/istio-cni.spec (hunk 1 above)
Comment (luksa) [reply istio-cni/istio-cni.spec:24]: Hmm, I just copy/pasted from one of the other RPMs. Does the comment still stand for el8?

File: istio-cni/istio-cni.spec (hunk 2 above)
Comment (luksa) [reply istio-cni/istio-cni.spec]: No, I forgot to remove it. Thanks.

File: istio-cni/istio-cni.spec (hunk 1 above)
Comment (luksa) [reply istio-cni/istio-cni.spec:24]: Oh, wait.. I don't even know what this does :D.. maybe el6 being mentioned here is ok :).

Comment (jwendell) [reply istio-cni/istio-cni.spec:24]: NM, this is there since the beginning.

File: istio-cni/istio-cni.spec [hunk 3]
```diff
@@ -0,0 +1,101 @@
+%undefine _missing_build_ids_terminate_build
+
+%global git_commit 13215cf36ca35df88549d3c6ba25b7330e1fa31d
```
Comment (jwendell) [inline istio-cni/istio-cni.spec]: This needs to be updated, probably.

File: istio-cni/istio-cni.spec [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+%files
+%license istio-cni/src/%{provider_prefix}/LICENSE
+%doc     istio-cni/src/%{provider_prefix}/README.md
+
+/opt/cni/bin/istio-cni
+/opt/cni/bin/istio-iptables.sh
+/install-cni.sh
+/filter.jq
+/istio-cni.conf.tmp
```
Comment (jwendell) [inline istio-cni/istio-cni.spec:96]: Are those files supposed to be in / ?

File: istio-cni/istio-cni.spec [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+
+%changelog
+* Fri Jun 14 2019 Marko Luksa <mluksa@redhat.com> - 0.1.0
+- First package
```
Comment (jwendell) [inline istio-cni/istio-cni.spec:101]: 0.12.0?

**Review by jwendell — CHANGES_REQUESTED** [review]
Made a few comments

File: istio-cni/update.sh [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    fi
+
+    sed -i "s/%global git_commit .*/%global git_commit ${sha}/" istio-cni.spec
+    NEW_SOURCES="${NEW_SOURCES} ${filename}"
+}
+
+function new_sources() {
+    echo
+    echo "Updating sources file with ${NEW_SOURCES}"
+    md5sum ${NEW_SOURCES} > sources
```
Comment (jwendell) [inline istio-cni/update.sh:45]: You don't seem to have a `sources` file...

File: istio-cni/istio-cni.spec (hunk 4 above)
Comment (luksa) [reply istio-cni/istio-cni.spec:96]: This is where they are in the upstream CNI image. We should probably move them to a more appropriate place, but I didn't know where that is, and I just wanted to get CNI up and running first (with as little change from upstream as possible).

File: istio-cni/istio-cni.spec (hunk 3 above)
Comment (luksa) [reply istio-cni/istio-cni.spec]: Done. Thanks.

File: istio-cni/istio-cni.spec (hunk 5 above)
Comment (luksa) [reply istio-cni/istio-cni.spec:101]: Thanks.

File: istio-cni/update.sh (hunk 6 above)
Comment (luksa) [reply istio-cni/update.sh:45]: I do have it: https://github.com/Maistra/rpms/pull/41/files#diff-6f26da552374a0e5211923d0837f13b5

Comment (jwendell) [reply istio-cni/update.sh:45]: Oh, sorry, haven't seen it. In this case, it is outdated :)

File: istio-cni/istio-cni.spec (hunk 1 above)
Comment (brian-avery) [reply istio-cni/istio-cni.spec:24]: Yeah, I don't see it being an issue. Just don't know if we have any assumptions about this being Red Hat Linux 6 based on the comment here. If it works elsewhere, feel free to ignore.

File: istio-cni/update.sh (hunk 6 above)
Comment (jwendell) [reply istio-cni/update.sh:45]: sorry, need a coffee

File: istio-cni/istio-cni.spec (hunk 4 above)
Comment (jwendell) [reply istio-cni/istio-cni.spec:96]: SGTM

**Review by jwendell — APPROVED** [review]
LGTM

**Pull request merged**